src/Sources/Coalescence.jl

```diff
@@ -3,425 +3,524 @@
+            ntuple(N) do k
+                get_normalized_kernel_tensor(kernel[j][k], norms)
             end
-            ind = get_dist_moment_ind(NProgMoms, k, m)
-            coal_data.coal_ints[ind] += sum(@views coal_data.Q[:, k])
-            coal_data.coal_ints[ind] -= sum(@views coal_data.R[:, k])
-            coal_data.coal_ints[ind] += coal_data.S[k, 1]
-            if k > 1
-                coal_data.coal_ints[ind] += coal_data.S[k - 1, 2]
+        end
+
+        N_mom_max = maximum(NProgMoms) + (P - 1)
+        N_2d_ints = ntuple(N) do i
+            if i < N
+                (P - 1) + max(NProgMoms[i], NProgMoms[i + 1])
+            else
+                (P - 1) + NProgMoms[i]
             end
         end
-    end
-end
 
-"""
-initialize_coalescence_data(
-    ::AnalyticalCoalStyle, 
-    kernel::Union{CoalescenceTensor{FT}, Matrix{CoalescenceTensor{FT}}},
-    NProgMoms;
-    dist_thresholds = nothing,)
+        thresholds = ntuple(N) do i
+            dist_thresholds[i] / norms[2]
+        end
 
-  - `kernel`: Array of kernel tensors that determine rate of coalescence based on pair of distributions and size of particles x, y
-  - `NProgMoms`: Array containing number of prognostic moments associated with each distribution
-  - `dist_thresholds`: PSD upper thresholds for computing S terms
-Initializes the coalescence data.
-"""
-function initialize_coalescence_data(
-    ::AnalyticalCoalStyle,
-    kernel::Union{CoalescenceTensor{N, FT, M}, Matrix{CoalescenceTensor{N, FT, M}}},
-    NProgMoms::Array{Int};
-    dist_thresholds = nothing,
-    norms = (FT(1), FT(1)),
-) where {N, M, FT <: Real}
-    Ndist = length(NProgMoms)
-    matrix_of_kernels = Array{CoalescenceTensor{N, FT, M}}(undef, Ndist, Ndist)
-    if kernel isa CoalescenceTensor
-        matrix_of_kernels .= get_normalized_kernel_tensor(kernel, norms)
-    else
-        @assert size(kernel) == (Ndist, Ndist)
-        for (k, ker) in enumerate(kernel)
-            matrix_of_kernels[k] = get_normalized_kernel_tensor(ker, norms)
+        new{N, P, FT, T}(N_mom_max, N_2d_ints, thresholds, kernels)
+    end
+
+    function CoalescenceData(
+        kernel::CoalescenceTensor{P, FT, T},
+        NProgMoms::NTuple{N, Int},
+        dist_thresholds::NTuple{N, FT},
+        norms::Tuple{FT, FT} = (FT(1), FT(1)),
+    ) where {N, P, T, FT <: Real}
+
+        kernel_ = get_normalized_kernel_tensor(kernel, norms)
+        kernels = ntuple(N) do j
+            ntuple(N) do k
+                kernel_
+            end
         end
+
+        CoalescenceData(kernels, NProgMoms, dist_thresholds, norms)
     end
 
-    Q = zeros(FT, Ndist, Ndist)
-    R = zeros(FT, Ndist, Ndist)
-    S = zeros(FT, Ndist, 2)
+end
 
-    kernels_order = [N for ker in matrix_of_kernels]
-    Nmom = maximum(NProgMoms) + maximum(kernels_order)
-    moments = zeros(FT, Ndist, Nmom)
+"""
+get_coal_ints(::AnalyticalCoalStyle, pdists::Array{ParticleDistribution{FT}}, coal_data::NamedTuple)
```

Comment:
```suggestion
    get_coal_ints(::AnalyticalCoalStyle, pdists::Array{ParticleDistribution{FT}}, coal_data::NamedTuple)
```
This should fix the formatting in the documentation, so that this function signature appears as code